uwblueprint/project-read-frontend
# Implement adding, editing, and deleting interactions

### Notion ticket link



[add, edit, and delete interactions](https://www.notion.so/uwblueprintexecs/Developer-Hub-30e9b6c82af24054b31acf7c5e822c89?p=a74c482b67d34348b4fb59cd2e1a53a3)



### Implementation description

very sorry this is a big PR i got a little carried away 😔 dm me for a walkthrough if you'd like one

- implemented an `InteractionCard` component that can be toggled to have edit mode to display a form
<img width="414" alt="Screen Shot 2021-08-08 at 5 39 01 PM" src="https://user-images.githubusercontent.com/37346399/128646479-14e1bf06-9fe8-428b-9488-3932d9c311e5.png">

- registered the PUT /family endpoint to call when an interaction gets saved or deleted



### Steps to test

1. run the backend on uwblueprint/project-read-backend#104
2. play around with the interactions, and ensure the behaviour is what you'd expect as a user!

https://user-images.githubusercontent.com/37346399/128646706-b2c0b9ec-a42b-494d-abc6-a5bed5684846.mov



### What should reviewers focus on?

- testing
- code quality & correctness
- have a good week 🌻 

### Checklist
